pkg/relay/ethereum/error_parse.go

```diff
@@ -0,0 +1,160 @@
+package ethereum
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+)
+
+type Artifact struct {
+	ABI []interface{} `json:"abi"`
+}
+
+type ErrorsRepository struct {
+	errs map[[4]byte]abi.Error
+}
+
+var erepo *ErrorsRepository
```

Comment:
Can we avoid using global variable here? I set it to ETHClient in ibc-solidity's test code. ref. https://github.com/hyperledger-labs/yui-ibc-solidity/blob/main/pkg/client/client.go#L45